implemented full restaurant module with CRUD and filters

keren-elbaz-m opened the pull request:
Created restaurant model with Joi validation schema. Built RestaurantRepository with custom methods for filter restaurants and extended restaurantController to include filter endpoints. Configured restaurant routes.

keren-elbaz-m committed:
implemented full restaurant module with CRUD and filters

Created restaurant model with Joi validation schema.
Built RestaurantRepository with custom methods for filter restaurants and extended restaurantController to include filter endpoints.
Configured restaurant routes.

DanielBaronMoveo commented on src/server.ts:
```diff
@@ -1,37 +1,43 @@
 app.use(express.json());
 
 app.use(ROUTES.API, healthRoute);
+app.use(`${ROUTES.API}${ROUTES.RESTAURANTS}`, restaurantRoutes);
```
Instead of repeating yourself here, let's create a prefix for our api routes, making it look something like the following: 

```
export const apiRouter = express.Router()
app.use('/api', apiRouter)

apiRouter.use('/chef', chefRoutes)
apiRouter.use('/restaurant', restaurantRoutes)
apiRouter.use('/dish', dishRoutes)

DanielBaronMoveo reviewed (approved):
Great job, let's male the minor adjustment there with the server and let's move on onwards the next resources (chefs, dishes)

keren-elbaz-m committed:
centralize API routes under apiRouter

Created apiRouter to group all /api routes and moved API prefix handling from server.ts to dedicated router
